### PR TITLE
Verify sig in MockERC1271Wallet

### PR DESCRIPTION
**Pull request by NIC619**
- Verify sig in `MockERC1271Wallet` so we can know at least if our signing utils in tests are correct

### CODE REVIEW

**Comment by NIC619** [comment]
So it turns out that ERC1271 wallet can't be used for PMM because 0x's signature does not match the wallet standard...

**Review by pilagod — APPROVED** [review]
Just curious about that is there any standard ruling the r, s, v order? 

I see EIP1271 doesn't mention any signature details. It seems to allow everyone to implement specific wallet, like (r, v, s), (s, r, v) 🤣

File: contracts/test/mocks/MockERC1271Wallet.sol [hunk 1]
```diff
@@ -37,10 +40,19 @@ contract MockERC1271Wallet is ISetAllowance, IERC1271Wallet {
...
+        require(operator == _ecrecover(keccak256(_data), _signature), "MockERC1271Wallet: invalid signature");
         return ERC1271_MAGICVALUE;
     }
 
     function isValidSignature(bytes32 _hash, bytes calldata _signature) external view override returns (bytes4 magicValue) {
+        require(operator == _ecrecover(_hash, _signature), "MockERC1271Wallet: invalid signature");
         return ERC1271_MAGICVALUE_BYTES32;
     }
+
+    function _ecrecover(bytes32 hash, bytes memory signature) internal pure returns (address) {
```
Comment (charlesjhongc) [inline contracts/test/mocks/MockERC1271Wallet.sol:52]: How about using OZ lib for this?
https://docs.openzeppelin.com/contracts/3.x/api/cryptography#ECDSA-recover-bytes32-bytes-

**Comment by NIC619** [comment]
> Just curious about that is there any standard ruling the r, s, v order?

Yeah there does not seem to be a standard. Can only presume people will follow popular library like OZ's.🧐

File: contracts/test/mocks/MockERC1271Wallet.sol (hunk 1 above)
Comment (NIC619) [reply contracts/test/mocks/MockERC1271Wallet.sol]: Updated in c369e28

File: contracts/test/mocks/MockERC1271Wallet.sol [hunk 2]
```diff
@@ -49,10 +50,10 @@ contract MockERC1271Wallet is ISetAllowance, IERC1271Wallet {
         return ERC1271_MAGICVALUE_BYTES32;
     }
 
-    function _ecrecover(bytes32 hash, bytes memory signature) internal pure returns (address) {
+    function _ecrecover(bytes32 _hash, bytes memory signature) internal pure returns (address) {
```
[the remark this answers is not in the document]
Comment (NIC619) [reply contracts/test/mocks/MockERC1271Wallet.sol]: Fixed in 7b38090

**Review by pilagod — DISMISSED** [review]

File: contracts/test/mocks/MockERC1271Wallet.sol (hunk 1 above)
Comment (NIC619) [reply contracts/test/mocks/MockERC1271Wallet.sol]: Replace `_ecrecover` with `ECDSA.recover` in 59fc2da

**Review by charlesjhongc — APPROVED** [review]

**Review by pilagod — APPROVED** [review]

**Pull request merged**